tests/client/net_handler_client.c

```diff
@@ -307,6 +307,12 @@ void start_net_handler() {
             printf("pthread_create: output dump\n");
         }
         usleep(400000);  // allow time for thread to dump output before returning to client
+        
+        // execute the keyboard_interface on a seperate thread
+        pthread_t keyboard_id; // id of thread running the keyboard_interface
+        if(pthread_create(&keyboard_id, NULL, (void *)setup_keyboard, NULL) != 0) {
+            printf("pthread create: setup keyboard");
+        } 
```

Comment:
Maybe put this in another function, because almost all the other logic in this function deals with setting up net handler itself